### PR TITLE
add and support is powering off sub message in battery status

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

**Comment by tridge** [comment]
I don't see a change to rename BATT_SMBUS_MEASUREMENT_INTERVAL_MS to BATT_SMBUS_MEASUREMENT_INTERVAL_US. I think that is needed?
Also, can you check that all other users of the battery_status message set is_powering_off to false?

**Comment by bugobliterator** [comment]
@tridge I've updated the PR, can you review?

File: src/drivers/batt_smbus/batt_smbus.cpp [hunk 1]
```diff
@@ -426,6 +430,27 @@ BATT_SMBUS::cycle()
...
+			if(_button_press_counts >= ((BATT_SMBUS_BUTTON_DEBOUNCE_MS * 1000) / BATT_SMBUS_MEASUREMENT_INTERVAL_US)) {
+				// battery will power off
+				new_report.is_powering_off = true;
+				// warn only once
+				if(_button_press_counts++ == ((BATT_SMBUS_BUTTON_DEBOUNCE_MS * 1000) / BATT_SMBUS_MEASUREMENT_INTERVAL_US)) {
+					warnx("system is shutting down NOW...");
+				}
+			} else if(pressed) {
+				// battery will power off if the button is held
+				_button_press_counts++;
```
Comment (tridge) [inline src/drivers/batt_smbus/batt_smbus.cpp]: what happens to the is_powering_off field for this case?

Comment (bugobliterator) [inline src/drivers/batt_smbus/batt_smbus.cpp]: @tridge the state will be whatever was the last state which will always be false, as the value is set to true only at one location which is when the only thing that's going to happen is shutdown.

Also if you are happy with this PR can you pull it in?

**Comment by tridge** [comment]
applied, thanks Sid!
please submit a PR to upstream for this change

**Pull request closed without merging**